Parse/PFObject.m

```diff
@@ -2317,6 +2317,31 @@ - (void)removeObjectForKey:(NSString *)key {
     }
 }
 
+- (void)revert {
+    @synchronized (lock) {
+        if ([self isDirty]) {
+            PFOperationSet *changes = [self unsavedChanges];
+            for (NSString *key in changes.keyEnumerator) {
+                [changes removeObjectForKey:key];
+            }
+            [self rebuildEstimatedData];
+            [_availableKeys removeAllObjects];
```

Comment:
It gets very tricky here, `_availableKeys` can contain keys that are in the serverData or in in-flight PFOperationSet (the one that is currently saving). I would suggest using all keys from estimated data + all keys from all operation sets (the reason for the latter, to avoid issue when the object was removed for any key and it would be gone from estimated data)